cmd/lncli/cmd_new_macaroon.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strings"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/urfave/cli"
+	"gopkg.in/macaroon.v2"
+)
+
+var newMacaroonCommand = cli.Command{
+	Name:     "newmacaroon",
+	Category: "Macaroons",
+	Usage: "Bakes a new macaroon with the provided list of " +
+		"permissions and restrictions",
+	ArgsUsage: "[--format= | --save_to=] [--timeout=] [--ip_address=] " +
+		"permissions",
+	Description: `
+	Create a new macaroon that grants the provided permissions and
+	optionally adds restrictions (timeout, IP address) to it.
+
+	The new macaroon can be shown on command line either in JSON
+	(--format=json) or hex (--format=hex) serialized format.
+	Or it can be saved directly to a file using the --save_to argument.
+
+	A permission is a tuple of an entity and an action, separated by a
+	colon. Multiple operations can be added as a comma separated list, for
+	example:
+
+	lncli newmacaroon info:read,invoices:write,foo:bar
+	`,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name: "format",
+			Usage: "the format to serialize and display the " +
+				"macaroon in on the command line, must be " +
+				"either 'json' or 'hex' (default 'hex')",
+		},
+		cli.StringFlag{
+			Name: "save_to",
+			Usage: "save the created macaroon to this file " +
+				"using the default binary format",
+		},
+		cli.Uint64Flag{
+			Name: "timeout",
+			Usage: "the number of seconds the macaroon will be " +
+				"valid before it times out",
+		},
+		cli.StringFlag{
+			Name:  "ip_address",
+			Usage: "the IP address the macaroon will be bound to",
+		},
+	},
+	Action: actionDecorator(newMacaroon),
+}
+
+func newMacaroon(ctx *cli.Context) error {
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	// Show command help if no arguments or flags are provided. Also show
+	// help if there are more arguments than we expect.
+	if (ctx.NArg() == 0 || ctx.NArg() > 1) && ctx.NumFlags() == 0 {
```

Comment:
`ctx.NArg() != 1` ?

Reply:
If there is one flag and no args, it won't show the help?

Reply:
Fixed.